lib/recurrence.ts

```
@@ -0,0 +1,131 @@
+function addOccurenceDate(baseDate: Date, frequency: RecurrenceFrequency, stepIndex: number): Date | null {
+    const base = new Date(baseDate.getTime())
+    base.setHours(0,0,0,0);
+    switch(frequency) {
+        case "daily":
+            return new Date(base.getFullYear(), base.getMonth(), base.getDate() + stepIndex);
+        case "weekly":
+            return new Date(base.getFullYear(), base.getMonth(), base.getDate() + stepIndex * 7);
+        case "biweekly":
+            return new Date(base.getFullYear(), base.getMonth(), base.getDate() + stepIndex * 14);
+        case "monthly":
+            const c = new Date(base.getFullYear(), base.getMonth() + stepIndex, base.getDate());
```

Biome (2.5.3)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

(lint/correctness/noSwitchDeclarations)